xgboost.ipynb

```diff
@@ -0,0 +1,227 @@
+    "\n",
+    "This example is similar but uses synthetic data instead, to allow it to be easily runnable. This example uses a Dask wrapper around XGBoost.\n",
+    "\n",
+    "<img src=\"http://dask.readthedocs.io/en/latest/_images/dask_horizontal.svg\" width=\"30%\" alt=\"Dask logo\"> <img src=\"https://raw.githubusercontent.com/dmlc/dmlc.github.io/master/img/logo-m/xgboost.png\" width=\"25%\" alt=\"Dask logo\">"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from dask import compute, persist\n",
+    "from dask.distributed import Client, progress\n",
+    "\n",
+    "client = Client(os.environ.get(\"DISTRIBUTED_ADDRESS\"))  # connect to cluster \n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First we create a bunch of synthetic data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask.array as da\n",
+    "from dask_ml.model_selection import train_test_split\n",
+    "import numpy as np\n",
+    "\n",
+    "n, d = int(100e3), 20\n",
+    "da.random.seed(42)\n",
+    "X = da.random.normal(size=(n, d), chunks=n // 100)\n",
+    "w_star = da.random.uniform(size=d, chunks=d)\n",
+    "w_star = w_star**3\n",
+    "y = da.sign(X @ w_star)\n",
+    "\n",
+    "# for binary:logistic objective, only [0, 1] labels allowed\n",
+    "y = (y + 1) / 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's separate into a training set and testing set, which will allow for good evaluation after training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.15)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's try and predict from this data with [dask-xgboost][dxgb]:\n",
+    "\n",
+    "[dxgb]:https://github.com/dask/dask-xgboost"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask_xgboost as dxgb\n",
+    "import xgboost as xgb"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "dask-xgboost is a small wrapper around xgboost, and will behave the same as xgboost.\n",
+    "\n",
+    "During training it will take care of handling the data, and allow xgboost to use it's own distributed scheduler."
```

Comment:
The first "it" is ambiguous here between Dask and XGBoost.  Lets use "Dask" to be explicit.  Lets also expand "handling" to "loading, cleaning, and pre-processing" or something similar.  Lets capitalize XGBoost to be polite to other software projects.  I recommend that we remove the term "scheduler", which is not commonly known by novices, and perhaps use something like "and leverage XGBoost's own distributed training system" instead.

We're not targetting people well versed in machine learning or distributed systems.  We're targetting readers who are far less sophisticated in this realm, but quite sophisticated in their own domain, like genomics or insurance.

Reply:
I've added

> XGBoost will leverage their own distributed training system using all the workers that Dask has available. Dask sets XGBoost up, gives XGBoost data and lets XGBoost do it's training in the background using all the workers Dask has available.

Does this clearly express what dask-xgboost does? I adopted the last line from [your blogpost on XGBoost][1]. Do I have your permission to use it here?

[1]:http://matthewrocklin.com/blog/work/2017/03/28/dask-xgboost